src/utils/featheryClient/index.ts

```diff
@@ -465,7 +509,11 @@ export default class FeatheryClient extends IntegrationClient {
       method: 'POST',
       body: formData
     };
-
+    // Here we can safely remove the listener becasue offlineRequestHandler has its own beforeunload
+    featheryWindow().removeEventListener(
```

Comment:
added: once pending changes cleared, stop flushing in case user tries to exit page